packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -0,0 +1,228 @@
+          );
+          resolve(undefined);
+        })
+        .catch((err: Error) => {
+          const errorMessage = `Unable to upload file(s) to AWS S3. Error ${err.message}`;
+          this.logger.error(errorMessage);
+          reject(errorMessage);
+        });
+    });
+  }
+
+  fetchTechDocsMetadata(entityName: EntityName): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
+
+      const fileStreamChunks: Array<any> = [];
+      this.storageClient
+        .headObject({
+          Bucket: this.bucketName,
+          Key: `${entityRootDir}/techdocs_metadata.json`,
+        })
+        .createReadStream()
+        .on('error', err => {
+          this.logger.error(err.message);
+          reject(err.message);
+        })
+        .on('data', chunk => {
+          fileStreamChunks.push(chunk);
+        })
+        .on('end', () => {
+          const techdocsMetadataJson = Buffer.concat(
+            fileStreamChunks,
+          ).toString();
+          resolve(techdocsMetadataJson);
+        });
+    });
+  }
+
+  /**
+   * Express route middleware to serve static files on a route in techdocs-backend.
+   */
+  docsRouter(): express.Handler {
+    return (req, res) => {
+      // Trim the leading forward slash
+      // filePath example - /default/Component/documented-component/index.html
+      const filePath = req.path.replace(/^\//, '');
+
+      // Files with different extensions (CSS, HTML) need to be served with different headers
+      const fileExtension = path.extname(filePath);
+      const responseHeaders = getHeadersForFileExtension(fileExtension);
+
+      const fileStreamChunks: Array<any> = [];
+      this.storageClient
+        .getObject({ Bucket: this.bucketName, Key: filePath })
+        .createReadStream()
+        .on('error', err => {
+          this.logger.warn(err.message);
+          res.status(404).send(err.message);
+        })
+        .on('data', chunk => {
+          fileStreamChunks.push(chunk);
+        })
+        .on('end', () => {
+          const fileContent = Buffer.concat(fileStreamChunks).toString();
+          // Inject response headers
+          for (const [headerKey, headerValue] of Object.entries(
+            responseHeaders,
+          )) {
+            res.setHeader(headerKey, headerValue);
+          }
+
+          res.send(fileContent);
+        });
+    };
+  }
+
+  /**
+   * A helper function which checks if index.html of an Entity's docs site is available. This
+   * can be used to verify if there are any pre-generated docs available to serve.
+   */
+  async hasDocsBeenGenerated(entity: Entity): Promise<boolean> {
+    return new Promise(resolve => {
+      const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
+      this.storageClient
+        .getObject({
```

Comment:
Since we only care if the file exists, instead of getting the object, we can do [`.headObject`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html) or just `check if object exists` if possible.